lib/widgets/set_status.dart

```diff
@@ -0,0 +1,339 @@
+                  ? null
+                  : handleStatusClear,
+              );
+            }),
+          ValueListenableBuilder(
+            valueListenable: statusChange,
+            builder: (_, change, _) {
+              return _ActionButton(
+                label: localizations.statusSaveButtonLabel,
+                icon: ZulipIcons.check,
+                onPressed: switch ((change.text, change.emoji)) {
+                  (OptionNone(), OptionNone()) => null,
+                  _                            => handleStatusSave,
+                });
+            }),
+        ],
+      ),
+      body: Column(children: [
+        Padding(
+          padding: const EdgeInsetsDirectional.only(
+            // In Figma design, this is 16px, but we compensate for that in
+            // the icon button below.
+            start: 8,
+            top: 8, end: 10,
+            // In Figma design, this is 4px, be we compensate for that in
+            // [SingleChildScrollView.padding] below.
+            bottom: 0),
+          child: Row(
+            crossAxisAlignment: CrossAxisAlignment.start,
+            children: [
+              IconButton(
+                onPressed: chooseStatusEmoji,
+                style: IconButton.styleFrom(
+                  splashFactory: NoSplash.splashFactory,
+                  foregroundColor: designVariables.icon,
+                  tapTargetSize: MaterialTapTargetSize.shrinkWrap,
+                  padding: EdgeInsets.symmetric(
+                    vertical: 8,
+                    // In Figma design, there is no horizontal padding, but we
+                    // provide it in order to create a proper tap target size.
+                    horizontal: 8)),
+                icon: Row(children: [
+                  ValueListenableBuilder(
+                    valueListenable: statusChange,
+                    builder: (_, change, _) {
+                      final emoji = change.emoji.or(oldStatus.emoji);
+                      return emoji == null
+                        ? const Icon(ZulipIcons.smile, size: 24)
+                        : UserStatusEmoji(emoji: emoji, size: 24, neverAnimate: false);
+                    }),
+                  SizedBox(width: 4),
+                  Icon(ZulipIcons.chevron_down, size: 16),
+                ]),
+              ),
+              Expanded(child: TextField(
+                controller: statusTextController,
+                minLines: 1,
+                maxLines: 2,
+                // The limit on the size of the status text is 60 characters.
+                // See: https://zulip.com/api/update-status#parameter-status_text
+                maxLength: 60,
+                cursorColor: designVariables.textInput,
+                textCapitalization: TextCapitalization.sentences,
+                style: TextStyle(fontSize: 19, height: 24 / 19),
+                decoration: InputDecoration(
+                  counterText: '',
```

Comment:
include a 1- or 2-line comment with a link to the thread about this (per https://github.com/zulip/zulip-flutter/pull/1701#discussion_r2216825570)